07_errors/runnerdave/storer_test.go

```diff
@@ -0,0 +1,206 @@
+package main
+
+import (
+	"testing"
+
+	tassert "github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+var (
+	puppy1 = func() Puppy {
+		return Puppy{ID: 11, Breed: "Chihuahua", Colour: "Brown", Value: 12.30}
+	}
+	puppy2 = func() Puppy {
+		return Puppy{ID: 12, Breed: "Cacri", Colour: "Undefined", Value: 1.30}
+	}
+	puppy3 = func() Puppy {
+		return Puppy{ID: 12, Breed: "Imaginary", Colour: "Undefined", Value: -1.30}
+	}
+)
+
+type storerSuite struct {
+	suite.Suite
+	store Storer
+}
+
+func (s *storerSuite) TestUpdatePuppyIDDoesNotExist() {
+	// given
+	assert := tassert.New(s.T())
+	testPuppy := puppy1()
+
+	// when
+	err := s.store.UpdatePuppy(13, &testPuppy)
+
+	// then
+	assert.Error(err, "Should error if id not found")
+	serr, ok := err.(*Error)
+	if ok {
+		assert.Equal(uint16(0x2), serr.Code)
+	}
+}
+
+func (s *storerSuite) TestUpdatePuppy() {
+	// given
+	assert := tassert.New(s.T())
+	testPuppy := puppy1()
+	targetPuppy := puppy2()
+	createError := s.store.CreatePuppy(&testPuppy)
+	r := require.New(s.T())
+	r.NoError(createError, "Create should not produce an error")
+
+	// when
+	uerr := s.store.UpdatePuppy(11, &targetPuppy)
```

Comment:
Slight inconsistency - you use `createError` above, `uerr` here and `err` and `deleteError` after. Best to be consistent. This also appears in other test functions.

Reply:
implemented